src/viewer.js

```diff
@@ -2983,7 +2992,13 @@ function onCanvasDrag( event ) {
 
     gestureSettings = this.gestureSettingsByDeviceType( event.pointerType );
 
-    if ( gestureSettings.dragToPan && !canvasDragEventArgs.preventDefaultAction && this.viewport ) {
+    if (gestureSettings.dblClickToZoom && THIS[ this.hash ].draggingToZoom){
+        var factor = Math.pow( this.zoomPerDblTapDrag, event.delta.y / 50 );
+        this.viewport.zoomBy(factor);
+        // this.viewport.applyConstraints();
+    }
+
+    if (gestureSettings.dragToPan && !THIS[ this.hash ].draggingToZoom && !canvasDragEventArgs.preventDefaultAction && this.viewport ) {
```

Comment:
Maybe this should be a `else if`? It doesn't really matter in terms of the logic, but it more strongly indicates the intent that only one should happen.